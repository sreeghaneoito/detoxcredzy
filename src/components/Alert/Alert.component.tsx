import {View, Text} from 'react-native';
import React from 'react';
import {style} from './Alert.style';
import {AlertProps} from '../../interfaces/component.interface';

/**
 * Alert Component
 * @param type: Alert type it will be warning, error or success
 * @param heading: Alert Heading
 * @param desc: Alert Description
 * @returns JSX component
 */
const Alert = ({type, desc, heading}: AlertProps) => {
  try {
    return (
      <View
        style={[
          style.container,
          type === 'warning' && style.warningBg,
          type === 'error' && style.errorBg,
          type === 'success' && style.successBg,
        ]}>
        <Text
          style={[
            style.heading,
            type === 'warning' && style.warningText,
            type === 'error' && style.errorText,
            type === 'success' && style.successText,
          ]}>
          {heading}
        </Text>
        <Text
          style={[
            style.desc,
            type === 'warning' && style.warningText,
            type === 'error' && style.errorText,
            type === 'success' && style.successText,
          ]}>
          {desc}
        </Text>
      </View>
    );
  } catch (error) {
    console.log('Alert Component Crash =>', error);
    return <View style={style.container} />;
  }
};

export default Alert;
