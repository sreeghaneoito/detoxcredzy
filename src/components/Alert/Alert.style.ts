import {StyleSheet} from 'react-native';

export const style = StyleSheet.create({
  container: {
    marginVertical: 20,
    paddingHorizontal: 16,
    paddingVertical: 24,
    borderRadius: 4,
  },
  heading: {
    textAlign: 'center',
    fontSize: 16,
    fontWeight: '700',
  },
  desc: {
    textAlign: 'center',
    fontSize: 16,
    fontWeight: '400',
    lineHeight: 24,
  },
  warningBg: {
    backgroundColor: '#B08500' + `${40}`,
  },
  successBg: {
    backgroundColor: '#28AF51' + `${40}`,
  },
  errorBg: {
    backgroundColor: '#E30E2C' + `${40}`,
  },
  successText: {
    color: '#28AF51',
  },
  warningText: {
    color: '#B08500',
  },
  errorText: {
    color: '#E30E2C',
  },
});
