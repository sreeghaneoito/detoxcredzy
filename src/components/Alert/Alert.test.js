import React from 'react';
import renderer from 'react-test-renderer';
import Alert from './Alert.component';

test('renders correctly', () => {
  const tree = renderer.create(<Alert />).toJSON();
  expect(tree).toMatchSnapshot();
});
