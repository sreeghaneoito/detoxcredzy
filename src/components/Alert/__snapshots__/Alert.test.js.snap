// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "borderRadius": 4,
        "marginVertical": 20,
        "paddingHorizontal": 16,
        "paddingVertical": 24,
      },
      false,
      false,
      false,
    ]
  }
>
  <Text
    style={
      Array [
        Object {
          "fontSize": 16,
          "fontWeight": "700",
          "textAlign": "center",
        },
        false,
        false,
        false,
      ]
    }
  />
  <Text
    style={
      Array [
        Object {
          "fontSize": 16,
          "fontWeight": "400",
          "lineHeight": 24,
          "textAlign": "center",
        },
        false,
        false,
        false,
      ]
    }
  />
</View>
`;
